### PR TITLE
Parser refactor

### PR DESCRIPTION
**Pull request by jjsjann123**
1. Fixes `Profiling rule & Fusion rule should be separated.` in #1181 Separating logic between profiling and fuser partitioning
2. Minor fixes to remove compiler warning

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/interface.cpp [hunk 1]
```diff
@@ -195,7 +200,7 @@ RegisterOperators size_eq_guard({
         // if we would ever return refined tensor, which would change aliasing
         // analysis, we should update aliasdb pass.
         [](const Node* node) -> Operation {
-          return [](Stack* stack) {
+          return [](Stack& stack) {
```
Comment (kevinstephano) [inline torch/csrc/jit/codegen/cuda/interface.cpp:203]: Did you mean to change this for this PR?  It doesn't seem related.

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/interface.cpp:203]: > Minor fixes to remove compiler warning

yes it's intended, the old signature has been deprecated.

**Review by kevinstephano — APPROVED** [review]
Everything looks as I would expect.  LGTM.
